### PR TITLE
Print error information when checking API

### PR DESCRIPTION
**Pull request by jochenehret**

### CODE REVIEW

**Review by ctlong — DISMISSED** [review]

File: tasks/ensure-api-healthy/main.go
Comment (davewalter) [inline tasks/ensure-api-healthy/main.go:40]: Should calling `Close()` be "defer'ed" before reading from resp.Body?


```golang
defer resp.Body.Close()
body, err := io.ReadAll(resp.Body)
if err == nil {
...
```

Comment (jochenehret) [reply tasks/ensure-api-healthy/main.go:40]: I tried, but it doesn't really work because we're inside a loop. So I applied this suggested solution: https://stackoverflow.com/a/45617791

Comment (davewalter) [reply tasks/ensure-api-healthy/main.go:40]: That makes sense. In that case, should it be moved out of the error check so that it is always called before continuing?

Comment (jochenehret) [reply tasks/ensure-api-healthy/main.go:40]: I think we only need to call `Close()` when we call `ReadAll(resp.Body)` AND that function did not return an error. So I would say that this is the only place for `Close()`.

Comment (davewalter) [reply tasks/ensure-api-healthy/main.go:40]: How about [using a function](https://stackoverflow.com/questions/45617758/proper-way-to-release-resources-with-defer-in-a-loop) (anonymous or named) to hold the logic in the loop? That way, each iteration could defer closing the response body before moving on to the next request.

**Comment by ctlong** [comment]
I actually preferred the original code but with the `Close()` call outside of the conditional, e.g.

```golang
if resp.StatusCode != 200 {
  numSuccessResponses = 0
  s := fmt.Sprintf("Received HTTP %d from the API, resetting...", resp.StatusCode)
  body, err := io.ReadAll(resp.Body)
  resp.Body.Close()
  if err == nil {
    s = fmt.Sprintf("Received HTTP %d response from the API: %s\nresetting...", resp.StatusCode, string(body))
  }
  fmt.Println(s)
  continue
}
```

**Comment by davewalter** [comment]
The [`http` package docs](https://pkg.go.dev/net/http) state:

> The client must close the response body when finished with it

This [stackoverflow discussion](https://stackoverflow.com/questions/23928983/defer-body-close-after-receiving-response) goes further, including the caveat _"whether you consume it or not"_ and states that not doing so will cause you to leak open file descriptors.

I _think_ this means that everything inside the timeout loop should be inside the anonymous function, and the defer statement should be right before the status code check to make sure it is called when the function exits.

**Comment by davewalter** [comment]
@jochenehret Carson and I have pulled the API polling out into a helper function. We think this makes the code more readable and allows it to defer the closing of the body response in a more obvious way. Please go ahead and merge this if it looks good to you (we couldn't add you as an approver).

**Review by davewalter — APPROVED** [review]

**Comment by jochenehret** [comment]
Ok, thanks! Merging now.

**Pull request merged**
